src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -81,141 +76,73 @@ public ExamAccessService(ExamRepository examRepository, ExerciseGroupRepository
 
         // Check that the exam is visible
         if (exam.getVisibleDate() != null && exam.getVisibleDate().isAfter(ZonedDateTime.now())) {
-            return forbidden();
+            return badRequest("exam.visibleDate", "400", "Can't access the exam before it is visible!");
         }
 
-        return ResponseEntity.ok(studentExam.get());
+        return ResponseEntity.ok(studentExam);
     }
 
     /**
-     * Checks if the current user is allowed to manage exams of the given course.
+     * Checks if the current user is allowed to manage exams of the given course, otherwise throws an AccessForbiddenException
```

Comment:
This function checks the course permission for given role, isn't it? it has nothing to with managing exam, right?